### PR TITLE
Feature/better ex handling

### PR DESCRIPTION
**Pull request by mpenet**
I might just throw away the default-* impls and have just 2 impl. of handlers with a default decoder that does `identity` instead. Still unsure. But that covers all the possible ways one might want to deal with exceptions (unchanged, with/without ex-info, throwing or not etc etc). 

I ll let it rest and see how I feel about it monday.

### CODE REVIEW

File: src/clj/qbits/spandex.clj [hunk 1]
```diff
@@ -202,30 +203,66 @@
                (response-headers response)
                (.getHost response))))
 
+(defn response-ex->response
```
Comment (rborer) [inline src/clj/qbits/spandex.clj:206]: I think this kind of utility function should be kept even if you decide to not keep the whole exception handling.

Comment (mpenet) [reply src/clj/qbits/spandex.clj:206]: I merged a modified/simplified version of this PR earlier :). including this function

Comment (rborer) [reply src/clj/qbits/spandex.clj:206]: Oops I didn't saw it, thanks :-)

Comment (mpenet) [reply src/clj/qbits/spandex.clj:206]: You can now overwrite exception-handling and make this automatic: 

```clj
(request c {... :exception-handler (fn [x] (throw (decode-exception x)))})
```

or if you want to have a map with status-code & co and only exceptions for networking issues & co : 
```clj
(request c {... :exception-handler decoding-exception-handler})
```

Comment (mpenet) [reply src/clj/qbits/spandex.clj:206]: naming is still subject to change, I am not happy with what's in master atm, but I am busy with other stuff right now :)

Comment (rborer) [reply src/clj/qbits/spandex.clj:206]: Nice!

Comment (mpenet) [reply src/clj/qbits/spandex.clj:206]: fyi https://github.com/mpenet/spandex/blob/master/CHANGELOG.md#030

**Pull request closed without merging**